src/test/net_tests.cpp

```diff
@@ -22,6 +22,7 @@
 
 #include <memory>
 #include <string>
+#include <vector>
```

Comment:
This line is left over from an earlier push and should be removed.

Other than that, I think this PR was finished, provided missing coverage+context, and just needed some review begging or sales effort.